airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -460,12 +486,172 @@ class Audience(GoogleAdsStream):
     Ad Group Ad Labels stream: https://developers.google.com/google-ads/api/fields/v11/ad_group_ad_label
     """
 
-    primary_key = ["audience.id"]
+    primary_key = ["customer.id", "audience.id"]
 
 
 class Labels(GoogleAdsStream):
     """
     Labels stream: https://developers.google.com/google-ads/api/fields/v14/label
     """
 
-    primary_key = ["label.id"]
+    primary_key = ["customer.id", "label.id"]
+
+
+class ChangeStatus(IncrementalGoogleAdsStream):
+    primary_key = None
+    cursor_field = "change_status.last_change_date_time"
+
+    def __init__(self, start_date: str = None, conversion_window_days: int = 14, end_date: str = None, **kwargs):
+        if start_date is None:
+            # default value of two month ago is used because Google Ads returns only data for last 90 days otherwise returns error
+            start_date = pendulum.now().subtract(months=2).to_date_string()
+        if end_date is None:
+            end_date = pendulum.now().add(days=1).to_date_string()
+        super().__init__(start_date, conversion_window_days, end_date, **kwargs)
+
+    def get_query(self, stream_slice: Mapping[str, Any]) -> str:
+        query = GoogleAds.convert_schema_into_query(
+            schema=self.get_json_schema(),
+            report_name=self.name,
+            from_date=stream_slice.get("start_date"),
+            to_date=stream_slice.get("end_date"),
+            cursor_field=self.cursor_field,
+            resource_type=stream_slice.get("resource_type"),
+            limit=10000,
+        )
+        return query
+
+
+class IncrementalEventsStream(GoogleAdsStream, IncrementalMixin, ABC):
```

Comment:
can you explain conceptually how the `IncrementalEventsStream` works and relates to other streams? How does this replicate deletes exactly? I'm having a hard time understanding how this achieves delete replication. Unit testing would also help me understand how this works.

Reply:
I added unit tests and an explanation of IncrementalEventsStream to the docs.

Reply:
Please add docs for this class

Reply:
Added docs